crates/red_knot_python_semantic/resources/mdtest/unreachable.md

```diff
@@ -248,7 +283,166 @@ def f():
+
+To start, we make sure that we do not emit a diagnostic in this simple case:
+
+```py
+import sys
+
+if sys.version_info >= (3, 11):
+    ExceptionGroup  # no error here
+```
+
+Similarly, if we negate the logic, we also emit no error:
+
+```py
+if sys.version_info < (3, 11):
+    pass
+else:
+    ExceptionGroup  # no error here
+```
+
+This also works for more complex `if`-`elif`-`else` chains:
+
+```py
+if sys.version_info >= (3, 13):
+    ExceptionGroup  # no error here
+elif sys.version_info >= (3, 12):
+    ExceptionGroup  # no error here
+elif sys.version_info >= (3, 11):
+    ExceptionGroup  # no error here
+elif sys.version_info >= (3, 10):
+    pass
+else:
+    pass
+```
+
+The same works for ternary expressions:
+
+```py
+class ExceptionGroupPolyfill: ...
+
+MyExceptionGroup1 = ExceptionGroup if sys.version_info >= (3, 11) else ExceptionGroupPolyfill
+MyExceptionGroup1 = ExceptionGroupPolyfill if sys.version_info < (3, 11) else ExceptionGroup
+```
+
+Due to short-circuiting, this also works for Boolean operators:
+
+```py
+sys.version_info >= (3, 11) and ExceptionGroup
+sys.version_info < (3, 11) or ExceptionGroup
+```
+
+And in `match` statements:
+
+```py
+reveal_type(sys.version_info.minor)  # revealed: Literal[10]
+
+match sys.version_info.minor:
+    case 13:
+        ExceptionGroup
+    case 12:
+        ExceptionGroup
+    case 11:
+        ExceptionGroup
+    case _:
+        pass
+```
+
+Terminal statements can also lead to unreachable code:
+
+```py
+def f():
+    if sys.version_info < (3, 11):
+        raise RuntimeError("this code only works for Python 3.11+")
+
+    ExceptionGroup
+```
+
+Finally, not that anyone would ever use it, but it also works for `while` loops:
+
+```py
+while sys.version_info >= (3, 11):
+    ExceptionGroup
+```
+
+### Silencing errors for actually unknown symbols
+
+We currently also silence diagnostics for symbols that are not actually defined anywhere. It is
+conceivable that this could be improved, but is not a priority for now.
+
+```py
+if False:
+    does_not_exist
+
+def f():
+    return
+    does_not_exist
+```
+
+### Attributes
+
+When attribute expressions appear in unreachable code, we should not emit `unresolved-attribute`
+diagnostics:
+
+```py
+import sys
+import builtins
+
+if sys.version_info >= (3, 11):
+    # TODO
+    # error: [unresolved-attribute]
+    builtins.ExceptionGroup
+```
+
+### Imports
+
+When import statements appear in unreachable code, we should not emit `unresolved-import`
+diagnostics:
+
+```py
+import sys
+
+if sys.version_info >= (3, 11):
+    # TODO
+    # error: [unresolved-import]
+    from builtins import ExceptionGroup
+
+    # TODO
+    # error: [unresolved-import]
+    import builtins.ExceptionGroup
+
+    # See https://docs.python.org/3/whatsnew/3.11.html#new-modules
+
+    # TODO
+    # error: [unresolved-import]
+    import tomllib
+
+    # TODO
+    # error: [unresolved-import]
+    import wsgiref.types
+```
```

Comment:
I will leave attributes and imports for a follow up. I want to get some feedback on this first.